src/Sylius/Component/Registry/ServiceRegistry.php

```diff
@@ -70,9 +70,9 @@ public function register(string $identifier, $service): void
             throw new \InvalidArgumentException(sprintf('%s needs to be an object, %s given.', ucfirst($this->context), gettype($service)));
         }
 
-        if (!in_array($this->interface, class_implements($service), true)) {
+        if (!is_a($service, $this->className, true)) {
```

Comment:
It should be possible to use `!$service instanceof $this->className` here then :)

Reply:
Indeed, since `$service` is necessarily instanciated.